models/certificate.js

```diff
@@ -171,10 +171,27 @@ module.exports = (sequelize, { STRING, BOOLEAN, INTEGER, TEXT, DECIMAL }) => {
     });
   };
 
+  Certificate.validateLongChannelId = function (name, claimId) {
+    return new Promise((resolve, reject) => {
+      this.findOne({
+        where: {name, claimId},
+      })
+      .then(result => {
+        if (!result) {
+          return resolve(NO_CHANNEL);
+        };
+        resolve(claimId);
+      })
+      .catch(error => {
+        reject(error);
+      });
+    });
+  };
+
   Certificate.getLongChannelId = function (channelName, channelId) {
     logger.debug(`getLongChannelId(${channelName}, ${channelId})`);
     if (channelId && (channelId.length === 40)) {  // if a full channel id is provided
-      return new Promise((resolve, reject) => resolve(channelId));
+      return this.validateLongChannelId(channelName, channelId);
     } else if (channelId && channelId.length < 40) {  // if a short channel id is provided
```

Comment:
Since 40 is used as a magic/special value both inside of this function and elsewhere, it should be a constant somewhere. Probably on the `Claim` class?